consensus/polybft/fsm.go

```diff
@@ -425,32 +411,37 @@ func (f *fsm) VerifyStateTransactions(transactions []*types.Transaction) error {
 }
 
 // Insert inserts the sealed proposal
-func (f *fsm) Insert(p *pbft.SealedProposal) error {
+func (f *fsm) Insert(proposal []byte, committedSeals []*messages.CommittedSeal) (*types.Block, error) {
+	newBlock := &types.Block{}
+	if err := newBlock.UnmarshalRLP(proposal); err != nil {
+		return nil, fmt.Errorf("cannot unmarshal proposal: %w", err)
+	}
+
 	// In this function we should try to return little to no errors since
 	// at this point everything we have to do is just commit something that
 	// we should have already computed beforehand.
-	extra, _ := GetIbftExtra(f.block.Block.Header.ExtraData)
+	extra, _ := GetIbftExtra(newBlock.Header.ExtraData)
 
 	// create map for faster access to indexes
-	nodeIDIndexMap := make(map[pbft.NodeID]int, f.validators.Len())
+	nodeIDIndexMap := make(map[types.Address]int, f.validators.Len())
 	for i, addr := range f.validators.Accounts().GetAddresses() {
-		nodeIDIndexMap[pbft.NodeID(addr.String())] = i
+		nodeIDIndexMap[addr] = i
 	}
 
 	// populated bitmap according to nodeId from validator set and committed seals
 	// also populate slice of signatures
 	bitmap := bitmap.Bitmap{}
-	signatures := make(bls.Signatures, 0, len(p.CommittedSeals))
+	signatures := make(bls.Signatures, 0, len(committedSeals))
 
-	for _, commSeal := range p.CommittedSeals {
-		index, exists := nodeIDIndexMap[commSeal.NodeID]
+	for _, commSeal := range committedSeals {
+		index, exists := nodeIDIndexMap[types.BytesToAddress(commSeal.Signer)]
```

Comment:
Extract this to a local variable (it is used twice)?
`types.BytesToAddress(commSeal.Signer)`